### PR TITLE
doc: implement integration

### PR DESCRIPTION
**Pull request by YLeight**
To build the documentation run `npm run doc`

Integrate jsDoc or similar documentation tool #1

### CODE REVIEW

**Comment by JIoJIaJIu** [comment]
Postpone till fix https://github.com/nodejs-tensorflow/nodejs-tensorflow/issues/34

File: package.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
     "download_libs": "grunt --gruntfile tools/Gruntfile download_libtensorflow --force",
     "gyp": "node-gyp configure build",
     "build": "webpack --config tools/webpack.config.js",
-    "test": "mocha-webpack --colors --webpack-config tools/webpack.config.test.js --full-trace \"test/**/*.js\""
+    "test": "mocha-webpack --colors --webpack-config tools/webpack.config.test.js --full-trace \"test/**/*.js\"",
+    "doc": "./node_modules/.bin/typedoc --out ./doc ./src"
```
Comment (JIoJIaJIu) [inline package.json]: 1. `./node_modules/bin` prefix is not needful cause `npm` by default populates `PATH` variable
2. There is already `docs` folder on the root path, having also `doc` will be strange
3. At this moment out path can be as `./doc/api` and add this path to .gitignore, please
4. There is [the doc repo](https://github.com/nodejs-tensorflow/nodejs-tensorflow-docs) where this logic should be in future, but it's not critical at this stage

**Comment by YLeight** [comment]
@JIoJIaJIu, done.

File: package.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
     "download_libs": "grunt --gruntfile tools/Gruntfile download_libtensorflow --force",
     "gyp": "node-gyp configure build",
     "build": "webpack --config tools/webpack.config.js",
-    "test": "mocha-webpack --colors --webpack-config tools/webpack.config.test.js --full-trace \"test/**/*.js\""
+    "test": "mocha-webpack --colors --webpack-config tools/webpack.config.test.js --full-trace \"test/**/*.js\"",
+    "doc": "typedoc --out ./doc/api ./src"
```
Comment (JIoJIaJIu) [inline package.json]: `--out ./docs/api..`

**Pull request closed without merging**